pkg/build/controller/config_controller.go

```
@@ -33,11 +37,32 @@ func IsFatal(err error) bool {
 
 type BuildConfigController struct {
 	BuildConfigInstantiator buildclient.BuildConfigInstantiator
+
+	KubeClient kclient.Interface
+	Client     osclient.Interface
 }
 
 func (c *BuildConfigController) HandleBuildConfig(bc *buildapi.BuildConfig) error {
 	glog.V(4).Infof("Handling BuildConfig %s/%s", bc.Namespace, bc.Name)
 
+	if strategy := bc.Spec.Strategy.JenkinsPipelineStrategy; strategy != nil {
+		glog.V(4).Infof("Detected Jenkins pipeline strategy in %s/%s build configuration", bc.Namespace, bc.Name)
+		if _, err := c.KubeClient.Services(bc.Namespace).Get(buildapi.JenkinsDefaultServiceName); err == nil {
```

So I feel this should return an error & allow retry controller to do its thing if the API server is uncontactable.